doukutsu.h

```diff
@@ -10,127 +10,136 @@ namespace Doukutsu {
-        // ???
-        StagePermitsStart = static_cast<uint32_t>(0x4A54FC),
-        // uint16_t[2]
-        MapSize = static_cast<uint32_t>(0x49E586),
-        // uint8_t[w * h]* : This is NOT collision data
-        MapPtr = static_cast<uint32_t>(0x49E480),
-        // uint8_t[256] : this is NOT a pointer!
-        MapPXA = static_cast<uint32_t>(0x49E484),
-        // NPCs
-        NPCBase = static_cast<uint32_t>(0x4A6220),
-        // NPC count
-        NPCCount = static_cast<uint32_t>(0x200),
-        NPCSize = static_cast<uint32_t>(0xAC)
+    enum Offsets : quint32 {
+        // quint8[5] : is 0xA1, 0x20, 0x8B, 0x49, 0x00 in Doukutsu.exe
+        VerifyExe = off(0x412429),
+        // quint32[0x20] : first element in item array
+        InventoryStart = off(0x499B40),
+        // Weapon[8] : first element in weapon array
+        WeaponsStart = off(0x499BC8),
+        // quint32 : index into Weapons
+        // CurWeaponOff = WeaponsStart + CurWeaponSlot * sizeof(Weapon)
+        CurWeaponSlot = off(0x499C68),
+        // quint16 : bitfield (one bit per equip)
+        Equips = off(0x49E650),
+        // qint32
+        PlayerX = off(0x49E654),
+        // qint32
+        PlayerY = off(0x49E658),
+        // qint32 : <UNI value - 0 for normal, 1 for Ironhead, 2 for no movement
+        PlayerUniverse = off(0x49E64C),
+        // qint32
+        PlayerXVel = off(0x49E66C),
+        // qint32
+        PlayerYVel = off(0x49E670),
+        // quint32 : timer for mercy invuln after taking damage
+        InvulnTimer = off(0x49E6C8),
+        // quint16
+        HealthCurrent = off(0x49E6CC),
+        // quint16
+        WhimStarNum = off(0x49E6CE),
+        // quint32
+        HealthMaximum = off(0x49E6D0),
+        // quint32 : used for yellow bar when taking damage
+        HealthDisplayed = off(0x49E6D4),
+        // quint32 ??? unknown unit
+        AirLeft = off(0x49E6DC),
+        // quint32
+        BoosterFuel = off(0x49E6E8),
+        // StagePermit[8] : first element in Arthur's House teleporter destination (StagePermit) array
+        StagePermitsStart = off(0x4A54FC),
+        // quint16[2]
+        MapSize = off(0x49E586),
+        // quint8[w * h]* : This is NOT collision data
+        MapPXMOff = off(0x49E480),
+        // quint8[256] : this is NOT a pointer!
+        MapPXA = off(0x49E484),
+        // NPC[0x200] : first element in NPC array
+        NPCsStart = off(0x4A6220),
+        // Bullet[0x40] : first element in bullet array
+        BulletsStart = off(0x499C98),
+
+        // a message to 20kdc:
+        // DON'T USE THIS ENUM FOR NON-OFFSETS!!!
+        // total size of NPC array
+        // this could probably be a "magic number" (since it's unlikely to change and this thing is probably already incompatible with txin's stuff)
+        //NPCCount = off(0x200),
+        // size of NPC struct?
+        // use sizeof(NPC) instead, it's probably somehow more efficent
+        //NPCSize = off(0xAC)
```

Comment:
If NPCSize is being abandoned (reasonable, but see note) it should be verified at some point that `sizeof(NPC) == 0xAC`...